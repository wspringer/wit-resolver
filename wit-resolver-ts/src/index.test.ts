import { Resolver } from "./index.js";
import { describe, it, expect } from "vitest";
describe("Resolver", () => {
  it("should allow you to resolve wit", () => {
    const resolver = new Resolver();
    resolver.register(
      "test.wit",
      `
      package local:test;
      interface foo { }
    `
    );
    const resolved = resolver.resolve();
    expect(resolved).toMatchSnapshot();
  });
});
