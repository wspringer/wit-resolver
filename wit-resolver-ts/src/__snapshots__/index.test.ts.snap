// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Resolver > should allow you to resolve wit 1`] = `"{"worlds":[],"interfaces":[{"name":"foo","types":{},"functions":{},"package":0}],"types":[],"packages":[{"name":"local:test","interfaces":{"foo":0},"worlds":{}}]}"`;
